In [674]:
## Import libraries
import pandas as pd
import numpy as np


In [675]:
owners_df = pd.read_csv("owners.csv")
pets_df = pd.read_csv("pets.csv")
procedures_df = pd.read_csv("procedures.csv")
procedure_details_df = pd.read_csv("procedure_details.csv")

In [676]:
owners_df.head()

OwnerID     Name     Surname          StreetAddress                 City  \
0     6049   Debbie    Metivier        315 Goff Avenue         Grand Rapids   
1     2863     John   Sebastian      3221 Perry Street              Davison   
2     3518   Connie      Pauley  1539 Cunningham Court  Bloomfield Township   
3     3663     Lena  Haliburton   4217 Twin Oaks Drive        Traverse City   
4     1070  Jessica   Velazquez   3861 Woodbridge Lane           Southfield   

  State StateFull  ZipCode  
0    MI  Michigan    49503  
1    MI  Michigan    48423  
2    MI  Michigan    48302  
3    MI  Michigan    49684  
4    MI  Michigan    48034

In [677]:
pets_df.head()

PetID     Name    Kind  Gender  Age  OwnerID
0  J6-8562   Brandy     Dog    male   11     5168
1  Q0-2001   Roomba     Cat    male    9     5508
2  M0-2904    Simba     Cat    male    1     3086
3  R3-7551   Keller  Parrot  female    2     7908
4  P2-7342  Cuddles     Dog    male   13     4378

In [678]:
procedures_df.head()


PetID   Date ProcedureType  ProcedureSubCode
0  T0-3277  42391  VACCINATIONS                 5
1  G6-6501  42398  VACCINATIONS                 5
2  P9-3625  42399    ORTHOPEDIC                 7
3  F1-1855  42402  VACCINATIONS                 5
4  S1-2243  42404  VACCINATIONS                 5

In [679]:
procedure_details_df.head()


ProcedureType  ProcedureSubCode  Description  Price
0   OFFICE FEES                 1  Office Call     32
1   OFFICE FEES                 2    Emergency    100
2   OFFICE FEES                 3         Reck     24
3      GROOMING                 1         Bath     15
4      GROOMING                 2     Flea Dip     15

In [680]:
print("Shape of owners Dataframe : ",owners_df.shape)
print("Shape of Pets Dataframe : ",pets_df.shape)
print("Shape of Procedure Dataframe : ",procedures_df.shape)
print("Shape of Procedure details Dataframe : ",procedure_details_df.shape)


Shape of owners Dataframe :  (89, 8)
Shape of Pets Dataframe :  (100, 6)
Shape of Procedure Dataframe :  (41, 4)
Shape of Procedure details Dataframe :  (41, 4)


In [681]:
owners_pets_joined = owners_df.join(pets_df.set_index('OwnerID'), on = "OwnerID" ,how = 'outer', lsuffix='_owners', rsuffix='_pets')

In [682]:
owners_pets_joined.head()

OwnerID Name_owners     Surname          StreetAddress  \
0     6049      Debbie    Metivier        315 Goff Avenue   
1     2863        John   Sebastian      3221 Perry Street   
2     3518      Connie      Pauley  1539 Cunningham Court   
3     3663        Lena  Haliburton   4217 Twin Oaks Drive   
4     1070     Jessica   Velazquez   3861 Woodbridge Lane   

                  City State StateFull  ZipCode    PetID Name_pets Kind  \
0         Grand Rapids    MI  Michigan    49503  I6-9459   Biscuit  Dog   
1              Davison    MI  Michigan    48423  R4-6131   Biscuit  Dog   
2  Bloomfield Township    MI  Michigan    48302  N6-7350   Biscuit  Dog   
3        Traverse City    MI  Michigan    49684  U4-6674   Biscuit  Dog   
4           Southfield    MI  Michigan    48034  U8-6473   Biscuit  Dog   

   Gender  Age  
0  female    4  
1  female    5  
2  female    8  
3  female    1  
4  female    3

In [683]:
len(owners_df["OwnerID"].unique())


89

In [684]:
len(pets_df['OwnerID'].unique())

89

In [685]:
owners_pets_joined.isnull().sum()

OwnerID          0
Name_owners      0
Surname          0
StreetAddress    0
City             0
State            0
StateFull        0
ZipCode          0
PetID            0
Name_pets        0
Kind             0
Gender           0
Age              0
dtype: int64

In [686]:
# owners_pets_joined.head()
# owners_pets_joined.to_csv("owners_pets_joined.csv")

# 1Q What is the name of the oldest dog in Southfield?

In [687]:

dogs_in_southfield_df = owners_pets_joined[(owners_pets_joined['City'] == 'Southfield') & (owners_pets_joined['Kind'] == 'Dog')]
dogs_in_southfield_df.head()

OwnerID Name_owners    Surname         StreetAddress        City State  \
4      1070     Jessica  Velazquez  3861 Woodbridge Lane  Southfield    MI   
32     8133      Stacey   Randolph         1145 D Street  Southfield    MI   
32     8133      Stacey   Randolph         1145 D Street  Southfield    MI   
34     7846       Elvia     Warren      2041 Eagle Drive  Southfield    MI   
34     7846       Elvia     Warren      2041 Eagle Drive  Southfield    MI   

   StateFull  ZipCode    PetID Name_pets Kind  Gender  Age  
4   Michigan    48034  U8-6473   Biscuit  Dog  female    3  
32  Michigan    48075  G0-7675       Taz  Dog    male    4  
32  Michigan    48075  O3-1895     Candy  Dog  female    3  
34  Michigan    48075  U4-9376     Scout  Dog  female    2  
34  Michigan    48075  H8-1429      Lily  Dog  female    3

In [688]:
# Method 1
# The row with the largest (oldest) value with respect to'Age' column from Southfield
Oldest_dog = oldest_dog_in_southfield = dogs_in_southfield_df.nlargest(1, 'Age')['Name_pets'].values[0]
#  Name of the oldest dog in Southfield?
print( 'The name of the oldest dog in Southfield :',Oldest_dog)

The name of the oldest dog in Southfield : Crockett


In [689]:
# method 2
dogs_df = dogs_in_southfield_df[dogs_in_southfield_df['Kind']=='Dog'].sort_values(by = 'Age', ascending = False).iloc[0]
# Details of the oldest dog -
print("The deatails of Oldest dog ")
dogs_df

The deatails of Oldest dog 


OwnerID                           1334
Name_owners                      Jason
Surname                       Cantwell
StreetAddress    2372 Don Jackson Lane
City                        Southfield
State                               MI
StateFull                     Michigan
ZipCode                          48075
PetID                          J0-7893
Name_pets                     Crockett
Kind                               Dog
Gender                            male
Age                                 12
Name: 43, dtype: object

## Ans: Name of the oldest pet in Southfield is :  Crockett

# 2Q What is the average (mean) number of pets per city

In [690]:
# Group by each city and counting the number of pets in each city
# Sort the values by the number_of_pets
count_per_city=owners_pets_joined.groupby('City').agg(number_of_pets=('PetID', 'count')).sort_values(by='number_of_pets', ascending =False)

# 
print(" The average (mean) number of pets per city : ", count_per_city['number_of_pets'].mean())

 The average (mean) number of pets per city :  2.5


## Ans : The average (mean) number of pets per city :  2.5

### Joining the procedureType and Procedure DataFrame

In [691]:
# Join on procedureType and Procedure Sub code
pet_proc_price_df = procedures_df.join(procedure_details_df.set_index(['ProcedureType','ProcedureSubCode']), on = ['ProcedureType','ProcedureSubCode'], how='left')
pet_proc_price_df.shape

(41, 6)

In [692]:
pet_proc_price_df.head()

PetID   Date ProcedureType  ProcedureSubCode   Description  Price
0  T0-3277  42391  VACCINATIONS                 5        Rabies     10
1  G6-6501  42398  VACCINATIONS                 5        Rabies     10
2  P9-3625  42399    ORTHOPEDIC                 7  Pinning-I.M.    325
3  F1-1855  42402  VACCINATIONS                 5        Rabies     10
4  S1-2243  42404  VACCINATIONS                 5        Rabies     10

In [693]:
len(owners_pets_joined['PetID'].unique())

100

# 3Q Which owner spent the most on procedures for their pet(s)?


In [694]:
# Merge the 'owners_pets_joined' DataFrame with the 'pet_proc_price_df' DataFrame 
# left join to have all the owners information
merged_df = owners_pets_joined.join(pet_proc_price_df.set_index('PetID'), 
                                                                 on = 'PetID', how = 'left')

In [695]:
merged_df.head()

OwnerID Name_owners     Surname          StreetAddress  \
0     6049      Debbie    Metivier        315 Goff Avenue   
1     2863        John   Sebastian      3221 Perry Street   
2     3518      Connie      Pauley  1539 Cunningham Court   
3     3663        Lena  Haliburton   4217 Twin Oaks Drive   
4     1070     Jessica   Velazquez   3861 Woodbridge Lane   

                  City State StateFull  ZipCode    PetID Name_pets Kind  \
0         Grand Rapids    MI  Michigan    49503  I6-9459   Biscuit  Dog   
1              Davison    MI  Michigan    48423  R4-6131   Biscuit  Dog   
2  Bloomfield Township    MI  Michigan    48302  N6-7350   Biscuit  Dog   
3        Traverse City    MI  Michigan    49684  U4-6674   Biscuit  Dog   
4           Southfield    MI  Michigan    48034  U8-6473   Biscuit  Dog   

   Gender  Age     Date ProcedureType  ProcedureSubCode Description  Price  
0  female    4      NaN           NaN               NaN         NaN    NaN  
1  female    5      NaN           NaN               NaN         NaN    NaN  
2  female    8      NaN           NaN               NaN         NaN    NaN  
3  female    1      NaN           NaN               NaN         NaN    NaN  
4  female    3  42561.0  VACCINATIONS               5.0      Rabies   10.0

In [696]:
merged_df.shape

(107, 18)

In [697]:
merged_df.isnull().sum()

OwnerID              0
Name_owners          0
Surname              0
StreetAddress        0
City                 0
State                0
StateFull            0
ZipCode              0
PetID                0
Name_pets            0
Kind                 0
Gender               0
Age                  0
Date                66
ProcedureType       66
ProcedureSubCode    66
Description         66
Price               66
dtype: int64

In [698]:
# Group by OwnerId and sum up with Price for each ownerID
# sort values by Price
merged_df1 = merged_df.groupby('OwnerID').agg({'Price':'sum'}).sort_values(by = 'Price', ascending = False)
merged_df1 = merged_df1.rename(columns = {'Price': 'Total_amount_spent'})
merged_df1.iloc[0]
#len(merged_df['Name_owners'].unique())

Total_amount_spent    450.0
Name: 8316, dtype: float64

In [637]:
merged_df1

Total_amount_spent
OwnerID                    
8316                  450.0
5806                  400.0
7579                  325.0
1915                  305.0
7581                  185.0
...                     ...
6406                    0.0
6731                    0.0
6923                    0.0
7023                    0.0
5737                    0.0

[89 rows x 1 columns]

In [699]:
# The highest amount spent is 450 and owner id is 8316 and using consition to look at the name
merged_df[merged_df['OwnerID'] == 8316]

OwnerID Name_owners Surname        StreetAddress          City State  \
70     8316      Daniel     Fay  894 Twin Oaks Drive  Grand Rapids    MI   

   StateFull  ZipCode    PetID Name_pets Kind Gender  Age     Date  \
70  Michigan    49503  J1-6366     Bruce  Dog   male    3  42665.0   

        ProcedureType  ProcedureSubCode         Description  Price  
70  GENERAL SURGERIES              17.0  Radical Mastectomy  450.0

## Ans: Owner spent most on procedure on pets is : Daniel Fay

# 4Q How many owners spent 20 dollars or more on procedures for their pets? 


In [700]:
# using the filter condition on merged dataframe with greater than 20 condition
Procedures_g20dollars = merged_df1[merged_df1['Total_amount_spent']>=20.0]

Procedures_g20dollars


Total_amount_spent
OwnerID                    
8316                  450.0
5806                  400.0
7579                  325.0
1915                  305.0
7581                  185.0
5168                  175.0
7261                  125.0
3089                   40.0
8143                   40.0
8215                   30.0
1070                   25.0
2809                   20.0

In [701]:
# len() gives the number of rows in the filterest dataframe
print(" The number of owners that spent 20 or more dollars is : ", len(Procedures_g20dollars))

 The number of owners that spent 20 or more dollars is :  12


## Ans: The number of owners that spent 20 or more dollars is :  12


# 5Q How many owners have at least two different kinds of pets (e.g. a dog and a cat)?


In [641]:
owners_with_unique_pets = owners_pets_joined.groupby('OwnerID')['Kind'].nunique()

owners_with_multiple_kinds = owners_with_unique_pets[owners_with_unique_pets >= 2]

owners_with_multiple_kinds
#owners_pets_dup[owners_pets_dup['OwnerID'].duplicated()]

OwnerID
3089    2
6102    2
7484    2
8133    2
Name: Kind, dtype: int64

In [642]:
Owners_multiple_petspecies = owners_pets_joined[owners_pets_joined['OwnerID'].isin([3089,6102,7484,8133])]

In [643]:
Owners_multiple_petspecies

OwnerID Name_owners    Surname         StreetAddress        City State  \
32     8133      Stacey   Randolph         1145 D Street  Southfield    MI   
32     8133      Stacey   Randolph         1145 D Street  Southfield    MI   
32     8133      Stacey   Randolph         1145 D Street  Southfield    MI   
58     6102      Robert  Partridge   704 Railroad Street   Marquette    MI   
58     6102      Robert  Partridge   704 Railroad Street   Marquette    MI   
69     3089         Lee   McKenzie  3295 Pinewood Avenue   Marquette    MI   
69     3089         Lee   McKenzie  3295 Pinewood Avenue   Marquette    MI   
69     3089         Lee   McKenzie  3295 Pinewood Avenue   Marquette    MI   
74     7484       Mario     Riddle      17 Rubaiyat Road   Fruitport    MI   
74     7484       Mario     Riddle      17 Rubaiyat Road   Fruitport    MI   

   StateFull  ZipCode    PetID Name_pets    Kind  Gender  Age  
32  Michigan    48075  G0-7675       Taz     Dog    male    4  
32  Michigan    48075  O3-1895     Candy     Dog  female    3  
32  Michigan    48075  T0-3277   Humbert     Cat    male   12  
58  Michigan    49855  H8-8856    Bandit  Parrot    male   11  
58  Michigan    49855  W8-5750     Simba     Cat    male   15  
69  Michigan    49855  L4-6292    Bright     Dog    male   13  
69  Michigan    49855  I7-9550     Angel     Dog  female   12  
69  Michigan    49855  G6-6501      Jake     Cat    male    2  
74  Michigan    49415  S4-4013       Pip     Dog    male    3  
74  Michigan    49415  O1-7229    Dexter  Parrot    male   10

In [644]:
print(" Owners that have at least two different kinds of pets : ", len(owners_with_multiple_kinds))

 Owners that have at least two different kinds of pets :  4


## Ans: Owners that have at least two different kinds of pets :  4

# 6Q How many owners have pets where the first letter of their name (OwnerName) matches the first letter of their pet's name (PetName)? E.g. Cookie and Charles.

In [645]:
# Convert 'Name_owners' and 'Name_pets' columns to lowercase for case-insensitive comparison
# compare the first letter of Name_owners and Name_pets with str[0]
matching_ownerspet_count = owners_pets_joined[owners_pets_joined['Name_owners'].str[0].str.lower() == owners_pets_joined['Name_pets'].str[0].str.lower()]
#matching_ownerspet_count

matching_ownerspet_count

OwnerID Name_owners    Surname          StreetAddress             City  \
5      7101      Bessie        Yen    30 Cunningham Court  Rochester Hills   
6      2419       Luisa    Cuellar   1308 Shingleton Road        Kalamazoo   
24     1653     Carolyn      Crane  3952 Jarvisville Road          Lansing   
29     3644     Dorothy  Whitworth     1348 Ripple Street          Saginaw   
36     7393         Tom     Topper        3518 Haven Lane      Grand Ledge   
53     5508     Charles     Swarey    2463 Charles Street            Flint   
65     6157         Dan     Hysell     3231 Hayhurst Lane         Plymouth   
73     2809       Bruce      Dunne      4119 Kimberly Way     Grand Rapids   

   State StateFull  ZipCode    PetID Name_pets Kind  Gender  Age  
5     MI  Michigan    48306  S4-0789   Biscuit  Dog  female   10  
6     MI  Michigan    49007  P0-1725      Lily  Dog  female    0  
24    MI  Michigan    48933  P9-6519    Cookie  Cat  female    6  
29    MI  Michigan    48607  W7-6372     Daisy  Dog  female    8  
36    MI  Michigan    48837  N8-0553     Tiger  Dog    male   15  
53    MI  Michigan    48548  F6-5391    Cookie  Cat  female    9  
65    MI  Michigan    48170  Q8-8753    Danger  Dog    male    7  
73    MI  Michigan    49503  K0-8177     Bruce  Dog    male   12

In [646]:
print("Owners matching the first letter of their Pets : ", len(matching_ownerspet_count['OwnerID'].unique()))


Owners matching the first letter of their Pets :  8


## Ans: Owners matching the first letter of their Pets :  8

# 7Q What percentage of pets received a vaccination?


In [647]:
merged_df.head()

OwnerID Name_owners     Surname          StreetAddress  \
0     6049      Debbie    Metivier        315 Goff Avenue   
1     2863        John   Sebastian      3221 Perry Street   
2     3518      Connie      Pauley  1539 Cunningham Court   
3     3663        Lena  Haliburton   4217 Twin Oaks Drive   
4     1070     Jessica   Velazquez   3861 Woodbridge Lane   

                  City State StateFull  ZipCode    PetID Name_pets Kind  \
0         Grand Rapids    MI  Michigan    49503  I6-9459   Biscuit  Dog   
1              Davison    MI  Michigan    48423  R4-6131   Biscuit  Dog   
2  Bloomfield Township    MI  Michigan    48302  N6-7350   Biscuit  Dog   
3        Traverse City    MI  Michigan    49684  U4-6674   Biscuit  Dog   
4           Southfield    MI  Michigan    48034  U8-6473   Biscuit  Dog   

   Gender  Age     Date ProcedureType  ProcedureSubCode Description  Price  
0  female    4      NaN           NaN               NaN         NaN    NaN  
1  female    5      NaN           NaN               NaN         NaN    NaN  
2  female    8      NaN           NaN               NaN         NaN    NaN  
3  female    1      NaN           NaN               NaN         NaN    NaN  
4  female    3  42561.0  VACCINATIONS               5.0      Rabies   10.0

In [648]:
# Filtering with VACCINATIONS 'ProcedureType'
Filtered_vaccine = merged_df[merged_df['ProcedureType'] == 'VACCINATIONS']

In [649]:
Filtered_vaccine.head()

OwnerID Name_owners    Surname         StreetAddress           City State  \
4      1070     Jessica  Velazquez  3861 Woodbridge Lane     Southfield    MI   
7      6194       Karen     Torres    3941 Ritter Avenue    Center Line    MI   
8      5833        Mary    Hurtado    4865 Juniper Drive  Saint Charles    MI   
10     7581    Florence      Nolen    3103 Howard Street   Grand Rapids    MI   
12     9900       Marie      Floyd  314 Cunningham Court     Southfield    MI   

   StateFull  ZipCode    PetID Name_pets    Kind  Gender  Age     Date  \
4   Michigan    48034  U8-6473   Biscuit     Dog  female    3  42561.0   
7   Michigan    48015  O6-3123   Biscuit     Dog  female    2  42624.0   
8   Michigan    48655  T0-5705   Biscuit     Dog  female    5  42678.0   
10  Michigan    49503  X0-8765   Vuitton  Parrot  female   11  42447.0   
12  Michigan    48075  I2-7926      Dior  Parrot  female    4  42518.0   

   ProcedureType  ProcedureSubCode Description  Price  
4   VACCINATIONS               5.0      Rabies   10.0  
7   VACCINATIONS               5.0      Rabies   10.0  
8   VACCINATIONS               5.0      Rabies   10.0  
10  VACCINATIONS               5.0      Rabies   10.0  
12  VACCINATIONS               5.0      Rabies   10.0

In [650]:
# Finding how many pet_id recieved the vaccination from the filtered 
len(Filtered_vaccine['PetID'].unique())

24

In [651]:
# Finding how many pet_ids are unique
len(merged_df['PetID'].unique())

100

In [652]:
## divding vaccinated number of pets with number of pets for percentage
percentage_of_vaccinated_pets = (len(Filtered_vaccine['PetID'].unique()) / len(merged_df['PetID'].unique())) * 100

print('percentage of pets received a vaccination  {:.2f}%'.format(percentage_of_vaccinated_pets))

percentage of pets received a vaccination  24.00%


## Ans: percentage of pets received a vaccination  24.00%

# 8Q What percentage of cities have more male pets than female pets?


In [653]:
## Grouping data with Cities having more males
grouped_data = owners_pets_joined.groupby(['City', 'Gender']).size().reset_index(name='Count').sort_values(by = 'City', ascending = True)


In [654]:
grouped_data.head()

City  Gender  Count
0            Ann Arbor    male      2
1  Bloomfield Township  female      3
2  Bloomfield Township    male      1
3             Bridgman    male      1
4       Bruce Crossing    male      1

In [655]:
# creating a pivot table and having male and female filled with repective counts
pivoted_data = grouped_data.pivot(index='City', columns='Gender', values='Count')
pivoted_data[['female','male']] = pivoted_data[['female','male']].astype('Int64')


In [656]:
pivoted_data.head()

Gender               female  male
City                             
Ann Arbor              <NA>     2
Bloomfield Township       3     1
Bridgman               <NA>     1
Bruce Crossing         <NA>     1
Caro                      1  <NA>

In [657]:
# filling the NA values with 0 for better representation 
pivoted_data.fillna(0,inplace=True)

In [658]:
pivoted_data.head()

Gender               female  male
City                             
Ann Arbor                 0     2
Bloomfield Township       3     1
Bridgman                  0     1
Bruce Crossing            0     1
Caro                      1     0

In [659]:
# creating additional column to mark '1' if male > female else label as "0"
pivoted_data['Male_greaterto_Female'] = np.where(pivoted_data['male']> pivoted_data['female'], 1, 0)

In [660]:
pivoted_data.head()

Gender               female  male  Male_greaterto_Female
City                                                    
Ann Arbor                 0     2                      1
Bloomfield Township       3     1                      0
Bridgman                  0     1                      1
Bruce Crossing            0     1                      1
Caro                      1     0                      0

In [661]:
# summing up the Male_greaterto_Female column to get how many cities have male pets greater than female
pivoted_data['Male_greaterto_Female'].sum() 

23

In [662]:
# getting count of unique city to calculate percentage
len(owners_pets_joined['City'].unique())

40

In [663]:
## Calculating the percentage
percentage_MgF_cities = round(pivoted_data['Male_greaterto_Female'].sum() / len(owners_pets_joined['City'].unique()) *100,2)
print("Cities having males greater than female: {:.2f}%".format(percentage_MgF_cities))

Cities having males greater than female: 57.50%


## Ans: Cities having males greater than female: 57.50%

# 9Q Which city's pet sample is made up of exactly 70% dogs?

In [672]:
# Group by data with respect to city and kind and count the occurances
city_dog = owners_pets_joined.groupby(['City', 'Kind']).size().reset_index(name='Count').sort_values(by = 'City', ascending = True)


In [673]:

city_dog.head()

City    Kind  Count
0            Ann Arbor     Dog      1
1            Ann Arbor  Parrot      1
2  Bloomfield Township     Cat      1
3  Bloomfield Township     Dog      2
4  Bloomfield Township  Parrot      1

In [666]:
# Pivoting the data with column kind to get the each pet count in cities
city_dog = city_dog.pivot(index = 'City',columns = 'Kind', values = 'Count' )

In [667]:
city_dog.fillna(0,inplace=True)
city_dog.head()

Kind                 Cat  Dog  Parrot
City                                 
Ann Arbor            0.0  1.0     1.0
Bloomfield Township  1.0  2.0     1.0
Bridgman             1.0  0.0     0.0
Bruce Crossing       0.0  1.0     0.0
Caro                 1.0  0.0     0.0

In [668]:
# create a column to have total pets in each city to calculate the dogs percentage
city_dog['Total_Pets'] = city_dog.sum(axis=1)
city_dog.head()

Kind                 Cat  Dog  Parrot  Total_Pets
City                                             
Ann Arbor            0.0  1.0     1.0         2.0
Bloomfield Township  1.0  2.0     1.0         4.0
Bridgman             1.0  0.0     0.0         1.0
Bruce Crossing       0.0  1.0     0.0         1.0
Caro                 1.0  0.0     0.0         1.0

In [669]:
# Calculate the percentage of dogs and total pets in city
city_dog['Percentage_Dogs'] = (city_dog['Dog'] / city_dog['Total_Pets'] * 100)
# using round off funtion to round the numerical to 2 decimals
city_dog['Percentage_Dogs'] = city_dog['Percentage_Dogs'].round(2)

In [671]:
# Filtering the column 'Percentage_Dogs' with 70 percentage
city_dog[city_dog['Percentage_Dogs']==70.00]

Kind          Cat  Dog  Parrot  Total_Pets  Percentage_Dogs
City                                                       
Grand Rapids  1.0  7.0     2.0        10.0             70.0

## Ans Grand Rapids city's pet sample is made up of exactly 70% dogs